unified-runtime/test/adapters/level_zero/enqueue_alloc.cpp

```diff
@@ -0,0 +1,685 @@
+                                           nullptr, nullptr));
+      }
+
+      ASSERT_SUCCESS(urQueueFinish(queues[idx]));
+    });
+  }
+
+  for (auto &thread : threads) {
+    thread.join();
+  }
+}
+
+TEST_P(urL0EnqueueAllocMultiQueueSameDeviceTest, SuccessDependantMt) {
+  const size_t allocSize = std::get<1>(this->GetParam()).allocSize;
+  const size_t iterations = std::get<1>(this->GetParam()).iterations;
+  const auto enqueueUSMAllocFunc =
+      std::get<1>(this->GetParam()).funcParams.enqueueUSMAllocFunc;
+  const auto checkUSMSupportFunc =
+      std::get<1>(this->GetParam()).funcParams.checkUSMSupportFunc;
+
+  ur_device_usm_access_capability_flags_t USMSupport = 0;
+  ASSERT_SUCCESS(checkUSMSupportFunc(device, USMSupport));
+  if (!(USMSupport & UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ACCESS)) {
+    GTEST_SKIP() << "Selected USM type is not supported.";
+  }
+
+  std::vector<std::thread> threads;
+  for (size_t qIdx = 0; qIdx < queues.size(); qIdx++) {
+    // Create a thread per queue.
+    threads.emplace_back([&, qIdx, iterations] {
+      std::vector<void *> ptrs(iterations);
+      std::vector<ur_event_handle_t> allocEvents(iterations);
+      std::vector<ur_event_handle_t> fillEvents(iterations);
+      std::vector<ur_event_handle_t> freeEvents(iterations);
+
+      ASSERT_SUCCESS(enqueueUSMAllocFunc(queues[qIdx], nullptr, allocSize,
+                                         nullptr, 0, nullptr, &ptrs[0],
+                                         &allocEvents[0]));
+      ASSERT_NE(ptrs[0], nullptr);
+      ASSERT_NE(allocEvents[0], nullptr);
+
+      uint8_t fillPattern = 0xAF;
+      ASSERT_SUCCESS(urEnqueueUSMFill(queues[qIdx], ptrs[0], 1, &fillPattern,
+                                      allocSize, 1, &allocEvents[0],
+                                      &fillEvents[0]));
+      ASSERT_NE(fillEvents[0], nullptr);
+
+      ASSERT_SUCCESS(urEnqueueUSMFreeExp(queues[qIdx], nullptr, ptrs[0], 1,
+                                         &fillEvents[0], &freeEvents[0]));
+      ASSERT_NE(freeEvents[0], nullptr);
+
+      for (size_t i = 1; i < iterations; i++) {
+        ASSERT_SUCCESS(enqueueUSMAllocFunc(queues[qIdx], nullptr, allocSize,
+                                           nullptr, 1, &freeEvents[i - 1],
+                                           &ptrs[i], &allocEvents[i]));
+        ASSERT_NE(ptrs[i], nullptr);
+        ASSERT_NE(allocEvents[i], nullptr);
+
+        // Access the memory.
+        ASSERT_SUCCESS(urEnqueueUSMFill(queues[qIdx], ptrs[i], 1, &fillPattern,
+                                        allocSize, 1, &allocEvents[i],
+                                        &fillEvents[i]));
+        ASSERT_NE(fillEvents[i], nullptr);
+
+        ASSERT_SUCCESS(urEnqueueUSMFreeExp(queues[qIdx], nullptr, ptrs[i], 1,
+                                           &fillEvents[i], &freeEvents[i]));
+        ASSERT_NE(freeEvents[i], nullptr);
+      }
```

Comment:
can you check whether the values we get are correct? This should be testing whether the events correctly introduce synchronization.

Reply:
I've added a `SuccessDependantSharedAcrossQueuesSyncValidate` test case that validates previous fill operation was finished after explicitly waiting for a different queue that had an alloc event enqueued with dependency on this fill operation.